File: app/src/main/java/de/uni_marburg/mathematik/ds/serval/model/comparators/MeasurementComparator.java
package de.uni_marburg.mathematik.ds.serval.model.comparators;

import java.util.Comparator;

import de.uni_marburg.mathematik.ds.serval.model.event.Measurement;

/**
 * Compares {@link Measurement measurements} based on their {@link Measurement#type type} and {@link
 * Measurement#value value}.
 */
public class MeasurementComparator implements Comparator<Measurement> {
    
    @Override
    public int compare(Measurement measurement, Measurement t1) {
        if (measurement.getType().equals(t1.getType())) {
            return measurement.getType().compareTo(measurement.getType());
        }
        return measurement.getValue().compareTo(t1.getValue());
    }
}

File: app/src/main/java/de/uni_marburg/mathematik/ds/serval/model/comparators/MeasurementComparator.kt
package de.uni_marburg.mathematik.ds.serval.model.comparators

import de.uni_marburg.mathematik.ds.serval.model.event.Event
import de.uni_marburg.mathematik.ds.serval.model.event.Measurement
import java.util.*

/** Compares [events][Event] based on their [measurements][Measurement]. **/
class MeasurementComparator : Comparator<Measurement> {

    override fun compare(measurement: Measurement, t1: Measurement): Int {
        return if (measurement.type == t1.type) {
            measurement.type.compareTo(measurement.type)
        } else measurement.value.compareTo(t1.value)
    }
}
